cmd/faucet/faucet.go

```diff
@@ -280,6 +280,57 @@ func (f *faucet) webHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write(f.index)
 }
 
+// captchaVerify handles CAPTCHA token and verification.
+func captchaVerify(conn *websocket.Conn, captchaToken, captchaSecret, captcha string) error {
+	if captchaToken == "" {
+		return nil
+	}
+	form := url.Values{}
+	form.Add("secret", captchaSecret)
+	form.Add("response", captcha)
+
+	res, err := http.PostForm("https://hcaptcha.com/siteverify", form)
+	if err != nil {
+		return sendError(&wsConn{conn: conn}, err)
+	}
+	defer res.Body.Close()
+
+	var result struct {
+		Success bool            `json:"success"`
+		Errors  json.RawMessage `json:"error-codes"`
+	}
+	err = json.NewDecoder(res.Body).Decode(&result)
+	if err != nil || !result.Success {
+		log.Warn("Captcha verification failed", "err", string(result.Errors))
+		return sendError(&wsConn{conn: conn}, errors.New("CAPTCHA verification failed"))
+	}
+	return nil
+}
+
+// getInitialStats gathers initial stats from the network to report.
+func getInitialStats(f *faucet) (*types.Header, *big.Int, uint64, error) {
```

Comment:
It should be dead loop, until faucet is ready, right? otherwise keep showing "Faucet offline"
```
for head == nil || balance == nil {
...
// sleep 3s
}
```

Reply:
it is on purpose in this PR, but I think we should keep the for loop